src/lib.rs

```diff
@@ -3100,3 +3161,24 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &str) -> Option<
         None
     }
 }
+
+fn which(tool: &PathBuf) -> Option<PathBuf> {
+    fn check_exe(exe: &mut PathBuf) -> bool {
+        let exe_ext = std::env::consts::EXE_EXTENSION;
+        exe.exists() || (!exe_ext.is_empty() && exe.set_extension(exe_ext) && exe.exists())
+    }
+
+    // If |tool| is not just one "word," assume it's an actual path...
+    if tool.components().count() > 1 {
+        let mut exe = PathBuf::from(tool);
+        return if check_exe(&mut exe) { Some(exe) } else { None };
+    }
+
+    // Loop through PATH entries searching for the |tool|.
+    env::var_os("PATH").as_ref().and_then(|path_entries| {
+        env::split_paths(path_entries).find_map(|path_entry| {
+            let mut exe = path_entry.join(tool);
+            return if check_exe(&mut exe) { Some(exe) } else { None };
```

Comment:
The `return` here can be elided (the last expression in a closure is an implicit return)

Reply:
It says "you might have meant to ... `return Some(exe);`"...

Reply:
Oh that probably means you need to remove the semicolon at the end of the line as well, changing it to:

```rust
if check_exe(&mut exe) { Some(exe) } else { None }
```

Reply:
Right, that's what I tried, and it did complain, 1.53.